R/mock2.R

```diff
@@ -4,16 +4,78 @@
 #' `r lifecycle::badge("experimental")`
 #'
 #' These functions represent a second attempt at bringing mocking to testthat,
-#' incorporating what we've learned from the mockr, mockery, and mockthat package.
+#' incorporating what we've learned from the mockr, mockery, and mockthat
+#' packages.
 #'
 #' `with_mocked_bindings()` and `local_mocked_bindings()` work by temporarily
 #' changing variable bindings in the namespace of namespace `.package`.
-#' Generally, it's only safe to mock packages that you own. If you mock other
-#' packages, we recommend using `skip_on_cran()` to avoid CRAN failures if the
-#' implementation changes.
+#' Generally, it's only safe to mock packages that you own.
 #'
 #' These functions do not currently affect registered S3 methods.
 #'
+#' # Use
+#'
+#' There are three places that the function you are trying to mock might
+#' come from: inside your package, imported from an external package via
+#' the `NAMESPACE`, called from an external package with `::`. The first
+#' two cases you can mock the same way. For example, take this code:
+#'
+#' ```R
+#' function_to_mocked <- function() {
+#'   another_function()
+#' }
+#' ```
+#'
+#' It doesn't matter whether you wrote `another_function()` or you've
+#' imported it from another package with `@import` or `@importFrom`,
+#' you mock it the same way:
```

Comment:
```suggestion
#' It doesn't matter whether `another_function()` is defined by your package or you've
#' imported it from a dependency with `@import` or `@importFrom`,
#' you mock it the same way:
```

Reply:
Maybe: "you mock it exactly as you call it"?

Reply:
Although since that's not the right advice for the `::` case, maybe that's a bad sentence to utter.